### ISSUE TITLE
fit_with_cache mode

### ISSUE THREAD

**Issue by mert-kurttutan** [PriorLabs/TabPFN#193]
### Describe the bug

Is fit_with_cache supported currently ?
I dont see any speed up with subsequent calls.
In the source code, it seems to be using None value and not storing any KV cache.

### Steps/Code to Reproduce

```python
import random

from sklearn.datasets import fetch_openml
import numpy as np

from itertools import product

import logging


RANDOM_SEED = 42


from sklearn.preprocessing import LabelEncoder

from sklearn.metrics import (
    roc_auc_score,
)
from sklearn.model_selection import train_test_split

from tabpfn import TabPFNClassifier

logger = logging.getLogger(__name__)
logging.basicConfig(filename='example.log', encoding='utf-8', level=logging.INFO)

def get_data_snythetic(num_samples: int, num_features: int, num_classes: int) -> tuple[np.ndarray, np.ndarray]:
    # Synthetic dataset: Predict class membership
    # Features: 10 continuous features
    # Samples: 1000 cases
    np.random.seed(RANDOM_SEED)
    # num_samples = 195
    # num_features = 22
    # num_classes = 2

    X = np.random.rand(num_samples, num_features)
    y = np.random.randint(0, num_classes, num_samples)
    return X, y

def run_tabpfn() -> None:
    num_sample_list = [100]
    num_feature_list = [2, 4, 8, 16, 32, 64, 128, 200, 300, 400, 500]
    num_class_list = [2]
    csv_string = "num_samples,num_features,num_classes,train_time,predict_time\n"
    for num_samples, num_features, num_classes in product(num_sample_list, num_feature_list, num_class_list):
        logger.info("------------------------------------------")
        logger.info(f"num_samples: {num_samples}, num_features: {num_features}, num_classes: {num_classes}")
        X, y = get_data_snythetic(num_samples, num_features, num_classes)
        logger.info(f"dims of data: {X.shape}, {y.shape}")

        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.20, random_state=6)
        import time
        t_start = time.time()
        # Train and evaluate TabPFN
        tabpfn_model = TabPFNClassifier(
            random_state=RANDOM_SEED,
            fit_mode="fit_with_cache",
        ).fit(X_train, y_train)
        t_train = time.time()
        logger.info(f"TabPFN Training Time: {t_train - t_start:.4f} seconds")
        y_pred = tabpfn_model.predict_proba(X_test)
        t_predict = time.time()
        logger.info(f"TabPFN Prediction Time: {t_predict - t_train:.4f} seconds")
        y_pred2 = tabpfn_model.predict_proba(X_test)
        t_predict2 = time.time()
        logger.info(f"TabPFN Prediction Time: {t_predict2 - t_predict:.4f} seconds")

        csv_string += f"{num_samples},{num_features},{num_classes},{t_train - t_start},{t_predict - t_train}\n"

    with open("tabpfn_speed.csv", "w") as f:
        f.write(csv_string)


if __name__ == "__main__":
    # profile()
    run_tabpfn()
```

### Versions

```shell
Main branch
```

**Comment by mert-kurttutan** [comment]
Another point: The online documentation does not mention fit_mode param of `TabPFNClassifier`. Is it outdated? If it is, it might be a good idea to automate the sync docs with source code in this repo

**Comment by mert-kurttutan** [comment]
@LeoGrin I would appreciate if you could check my claim on this issue (about the performance for fit_with_cache)

**Comment by LeoGrin** [comment]
Thanks @mert-kurttutan !
Regarding your experiment, you should not expect the second predict call to be faster with `fit_with_cache`, rather you should expect the `fit` call to be slower than with other `fit_mode` and the `predict` call to be faster. The `fit_with_cache` option means that the kv cache  values of the train set will be cached during fit and reused during predict, but the test set is never cached. (I may have misunderstood your experiment).
Regarding the documentation, I see that it does not contain `TabPFNClassifier`'s parameters, I will open a new issue for this.

**Comment by LeoGrin** [comment]
Closing as I think the only remaining issue is the docs improvement in #195 , but feel free to reopen if you disagree @mert-kurttutan :)
